featurebyte/service/target.py

```diff
@@ -167,3 +180,63 @@ async def create_document(self, data: TargetCreate) -> TargetModel:
                     ),
                 )
         return await self.get_document(document_id=insert_id)
+
+    async def compute_targets(
+        self,
+        observation_set: Union[pd.DataFrame, ObservationTableModel],
+        compute_target_request: ComputeTargetRequest,
+        get_credential: Any,
+        output_table_details: TableDetails,
+        progress_callback: Optional[Callable[[int, str], None]] = None,
+    ) -> None:
+        """
+        Get target values for a target
+
+        Parameters
+        ----------
+        observation_set: pd.DataFrame
+            Observation set data
+        compute_target_request: ComputeTargetRequest
+            Compute target request
+        get_credential: Any
+            Get credential handler function
+        output_table_details: TableDetails
+            Table details to write the results to
+        progress_callback: Optional[Callable[[int, str], None]]
+            Optional progress callback function
+        """
+        feature_store = await self.feature_store_service.get_document(
+            document_id=compute_target_request.feature_store_id
+        )
+
+        if isinstance(observation_set, pd.DataFrame):
+            request_column_names = set(observation_set.columns)
+        else:
+            request_column_names = {col.name for col in observation_set.columns_info}
+
+        parent_serving_preparation = (
+            await self.entity_validation_service.validate_entities_or_prepare_for_parent_serving(
+                graph=compute_target_request.graph,
+                nodes=compute_target_request.nodes,
+                request_column_names=request_column_names,
+                feature_store=feature_store,
+                serving_names_mapping=compute_target_request.serving_names_mapping,
+            )
+        )
+
+        db_session = await self.session_manager_service.get_feature_store_session(
+            feature_store=feature_store,
+            get_credential=get_credential,
+        )
+
+        await get_targets(
+            session=db_session,
+            graph=compute_target_request.graph,
+            nodes=compute_target_request.nodes,
+            observation_set=observation_set,
+            serving_names_mapping=compute_target_request.serving_names_mapping,
+            feature_store=feature_store,
+            parent_serving_preparation=parent_serving_preparation,
+            output_table_details=output_table_details,
+            progress_callback=progress_callback,
+        )
```

Comment:
This is still in base document service...

Reply:
yea i thought you were ok with me moving this `compute_targets` function in a follow-up PR? did i understand wrongly =\

https://github.com/featurebyte/featurebyte/pull/1505#discussion_r1259098588

Reply:
i thought you mean moving to a file first (but not copy) before refactoring into a class.

Reply:
I misread this part. It is indeed a move but not copy. Since this is not used in other place, it should be ok now.